wgpu-hal/src/vulkan/device.rs

```diff
@@ -1012,10 +1016,53 @@ impl crate::Device<super::Api> for super::Device {
             })
             .collect::<Vec<_>>();
 
+        let dsl_create_flags = if self.shared.private_caps.update_after_bind {
+            vk::DescriptorSetLayoutCreateFlags::UPDATE_AFTER_BIND_POOL_EXT
+        } else {
+            vk::DescriptorSetLayoutCreateFlags::empty()
+        };
+
         let vk_info = vk::DescriptorSetLayoutCreateInfo::builder()
-            .flags(vk::DescriptorSetLayoutCreateFlags::empty())
+            .flags(dsl_create_flags)
             .bindings(&vk_bindings);
 
+        let mut binding_flag_info;
+        let binding_flag_vec;
+
+        let partially_bound = self
+            .shared
+            .features
+            .intersects(wgt::Features::PARTIALLY_BOUND_BINDING_ARRAY);
+
+        let vk_info = if self.shared.private_caps.update_after_bind || partially_bound {
+            binding_flag_vec = desc
+                .entries
+                .iter()
+                .map(|entry| {
+                    let mut flags = vk::DescriptorBindingFlags::empty();
+
+                    if partially_bound && entry.count.is_some() {
+                        flags |= vk::DescriptorBindingFlags::PARTIALLY_BOUND;
```

Comment:
We need to leave a note somewhere in wgpu-hal API to consider lifting this. This wouldn't make any difference to the end wgpu-rs users tho.

Reply:
wym by this?

Reply:
like, whether "partially bound" feature also means that wgpu-hal users specify a flag at BGL creation